backend/module/eCampCore/src/EntityService/CampCollaborationService.php

```diff
@@ -141,19 +179,26 @@ protected function updateEntity(BaseEntity $entity, $data) {
     protected function deleteEntity(BaseEntity $entity) {
         /** @var CampCollaboration $campCollaboration */
         $campCollaboration = $entity;
-        $data = (object) ['status' => CampCollaboration::STATUS_LEFT];
-
-        if ($campCollaboration->isEstablished()) {
-            $campCollaboration = $this->updateCollaboration($campCollaboration, $data);
-        } elseif ($campCollaboration->isInvitation()) {
-            $campCollaboration = $this->updateInvitation($campCollaboration, $data);
-        } elseif ($campCollaboration->isRequest()) {
-            $campCollaboration = $this->updateRequest($campCollaboration, $data);
+        if (in_array(
+            $campCollaboration->getStatus(),
+            [CampCollaboration::STATUS_INVITED, CampCollaboration::STATUS_REQUESTED],
+            true
+        )) {
+            parent::deleteEntity($entity);
+        } else {
+            $this->updateEntity($campCollaboration, (object) ['status' => CampCollaboration::STATUS_LEFT]);
         }
     }
 
     protected function fetchAllQueryBuilder($params = []) {
         $q = parent::fetchAllQueryBuilder($params);
+
+        if (isset($params['inviteKey'])) {
+            $q->orWhere('row.inviteKey = :inviteKey');
```

Comment:
Should this be `andWhere`?

Reply:
Doesn't really matter because it's the only condition.
I removed the default filter of the camp Entity that a not authenticated visitor can query the inviteKey.
Do you prefer andWhere?

Reply:
I'd prefer `andWhere` here, in case someone adds more filter options in the future and decides to put that code above your `inviteKey` code.

Reply:
Is now removed and solved with the RPC Controller